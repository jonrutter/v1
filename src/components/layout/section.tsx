import React from 'react';
import clsx from 'clsx';

// types
import type { PolymorphicProps } from '@/utils/polymorphic';

type Props = {
  className?: string;
};

/**
 * Renders a block of content, styled with the site's default spacing for consistency.
 *
 * Should be used as a direct child of `<Layout>`, in order to maintain correct site layout.
 */
export const Section = <T extends React.ElementType = 'section'>({
  as,
  className = '',
  children,
}: PolymorphicProps<T, Props>) => {
  const Tag = as || 'section';
  return (
    <div
      className={clsx(
        'py-12 sm:py-16 md:py-20 px-6 md:px-12 transition-all bg-inherit text-inherit',
        className
      )}
    >
      <Tag className="max-w-site-sm md:max-w-site-md lg:max-w-site-lg mx-auto">
        {children}
      </Tag>
    </div>
  );
};
