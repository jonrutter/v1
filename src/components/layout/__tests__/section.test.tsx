import React from 'react';
import renderer from 'react-test-renderer';

import { Section } from '../section';

const Component = <Section>Hello, world!</Section>;

describe('Section layout component', () => {
  it('renders the correct content', () => {
    const tree = renderer.create(Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
