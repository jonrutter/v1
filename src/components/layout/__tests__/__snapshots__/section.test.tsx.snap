// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section layout component renders the correct content 1`] = `
<div
  className="py-12 sm:py-16 md:py-20 px-6 md:px-12 transition-all bg-inherit text-inherit"
>
  <section
    className="max-w-site-sm md:max-w-site-md lg:max-w-site-lg mx-auto"
  >
    Hello, world!
  </section>
</div>
`;
